src/traces/splom/defaults.js

```diff
@@ -158,9 +158,10 @@ function handleAxisDefaults(traceIn, traceOut, layout, coerce) {
         }
     }
 
-    // when lower half is omitted, override grid default
-    // to make sure axes remain on the left/bottom of the plot area
-    if(!showLower) {
+    // when lower half is omitted, or when just the diagonal is gone,
+    // override grid default to make sure axes remain on
+    // the left/bottom of the plot area
+    if(!showLower || (!showDiag && showUpper && showLower)) {
```

Comment:
I could imagine `splom` *always* setting the defaults to `'bottom'` and `'left'`... but I guess in principle this way at least if you're in one of the situations that doesn't need this, you could make a bigger grid than the splom and put other things in other cells without having to tweak the axis positions. So yeah, nice fix 🎉

Reply:
> I could imagine `splom` _always_ setting the defaults to `'bottom'` and `'left'`

Yeah, that's how it was initially, but https://github.com/plotly/plotly.js/commit/c947b2c0070833199ceff56fc4bafc59d539f3d1#diff-f4f66cea15079ea63f9be7a90528570c added this block for perf benefits.